File: LinkedList/test_class.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package linkedlist;

/**
 *
 * @author Kakatuahitam
 */
public class test_class {
    
    
    public static void main(String[] args) {
        SingleLinkedList single_link = new SingleLinkedList();
        DoubleLinkedList double_link = new DoubleLinkedList();
        
        single_link.addEnd(10);
        double_link.addEnd(10);
        
        System.out.println(single_link.toString() + single_link.size());
        System.out.println(double_link.toString() + double_link.size());
        
        single_link.addAfter(10, 89);
        double_link.addAfter(10, 89);
        
        System.out.println(single_link.toString() + single_link.size());
        System.out.println(double_link.toString() + double_link.size());
        
        single_link.addBefore(89, 88);
        double_link.addBefore(89, 88);
        
        System.out.println(single_link.toString() + single_link.size());
        System.out.println(double_link.toString() + double_link.size());
        
        single_link.addAfter(88, 1);
        double_link.addAfter(88, 1);
        
        System.out.println(single_link.toString() + single_link.size());
        System.out.println(double_link.toString() + double_link.size());
        
        single_link.remove(89);
        double_link.remove(89);
        
        System.out.println(single_link.toString() + single_link.size());
        System.out.println(double_link.toString() + double_link.size());
        
        single_link.addBefore(10, "head");
        double_link.addBefore(10, "head");
        
        System.out.println(single_link.toString() + single_link.size());
        System.out.println(double_link.toString() + double_link.size());
        
        single_link.addBefore(1, "last-1");
        double_link.addBefore(1, "last-1");
        
        System.out.println(single_link.toString() + single_link.size());
        System.out.println(double_link.toString() + double_link.size());
    }
}

File: LinkedList/test_class.py
# -*- coding: utf-8 -*-
"""
Created on Tue Jun  2 04:31:00 2020

@author: Kakatuahitam
"""


from SingleLinkedList import SingleLinkedList as sll
from DoubleLinkedList import DoubleLinkedList as dll

s_list = sll()
d_list = dll()

s_list.addEnd(10)
d_list.addEnd(10)
print(s_list, s_list.wSize())
print(d_list, d_list.wSize())

s_list.addAfter(10, 89)
d_list.addAfter(10, 89)
print(s_list, s_list.wSize())
print(d_list, d_list.wSize())

s_list.addBefore(89, 88)
d_list.addBefore(89, 88)
print(s_list, s_list.wSize())
print(d_list, d_list.wSize())

s_list.addAfter(88, 1)
d_list.addAfter(88, 1)
print(s_list, s_list.wSize())
print(d_list, d_list.wSize())

s_list.remove(89)
d_list.remove(89)
print(s_list, s_list.wSize())
print(d_list, d_list.wSize())

s_list.addBefore(10, "head")
d_list.addBefore(10, "head")
print(s_list, s_list.wSize())
print(d_list, d_list.wSize())

s_list.addBefore(1, "last-1")
d_list.addBefore(1, "last-1")
print(s_list, s_list.wSize())
print(d_list, d_list.wSize())

# %%
if not(False == False):
    print("false")